app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/screen/TeleconsultRecordEffectHandler.kt

```diff
@@ -23,9 +32,36 @@ class TeleconsultRecordEffectHandler @AssistedInject constructor(
         .addAction(GoBack::class.java, uiActions::goBackToPreviousScreen)
         .addConsumer(NavigateToTeleconsultSuccess::class.java, { uiActions.navigateToTeleconsultSuccessScreen(it.teleconsultRecordId) }, schedulersProvider.ui())
         .addTransformer(LoadTeleconsultRecordWithPrescribedDrugs::class.java, loadTeleconsultRecordWithPrescribedDrugs())
+        .addTransformer(CreateTeleconsultRecord::class.java, createTeleconsultRecord())
         .build()
   }
 
+  private fun createTeleconsultRecord(): ObservableTransformer<CreateTeleconsultRecord, TeleconsultRecordEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map { effect ->
+            teleconsultRecordRepository.createTeleconsultRecordForMedicalOfficer(
+                teleconsultRecordId = uuidGenerator.v4(),
+                patientUuid = effect.patientUuid,
+                medicalOfficerId = user.get().uuid,
+                teleconsultRecordInfo = createTeleconsultRecordInfo(effect)
+            )
+          }
+          .map { TeleconsultRecordCreated }
+    }
+  }
+
+  private fun createTeleconsultRecordInfo(it: CreateTeleconsultRecord): TeleconsultRecordInfo {
+    return TeleconsultRecordInfo(
+        recordedAt = Instant.now(utcClock),
```

Comment:
There might be a couple of millisecond difference between this timestamp and the one used to create the timestamps in `TeleconsultRecordRepository#createTeleconsultRecordForMedicalOfficer`.

Since we are going to be merging multiple teleconsult records on the server side for reports, can we confirm with the server team whether they will depend on these timestamps being the same in order to merge them or select which records to use?

Reply:
The conversation is happening [here](https://simpledotorg.slack.com/archives/CV9V1DWLV/p1599472445007200).